universe/UniverseGenerator.cpp

```diff
@@ -172,7 +174,7 @@ namespace Delauney {
 
         // loop through points generated from systems, excluding the final
         // 3 points added for the covering triangle
-        for (int n = 0; n < points_vec.size() - 3; n++) {
+        for (std::size_t n = 0; n < points_vec.size() - 3 && n < std::numeric_limits<int>::max() ; n++) {
```

Comment:
Just `static_cast<int>(points_vec.size())` to avoid the extra casts later?

Is testing against `std::numeric_limits` really necessary?

Reply:
You can cast `size_t` to `int`, but not safely.  `int` is signed and `size_t` is unsigned and the spec places no restrictions on their relative sizes.

An alternative is to put the test before the loop and short circuit the loop if  `points_vec.size() > std::numeric_limits<int>::max()`.  However, the method in this PR allows the most possible amount of work to be done before it fails.

Realistically, something else will fail before it hits this limit.

Reply:
I would rather have tests earlier in the function to ensure that the input data is acceptable (eg. not more than 10^5 systems to triangulate) than have the extra tests cluttering the important bits of code. There is no benefit to looping over just some of the points.